spec/mailers/user_mailer_spec.rb

```diff
@@ -432,4 +468,77 @@
       end
     end
   end
+
+  describe "invited to collection" do
+    let!(:gift_exchange) { create(:gift_exchange) }
+    let!(:collection) { create(:collection, challenge: gift_exchange, challenge_type: "GiftExchange") }
+    let!(:otheruser) { create(:user) }
+    let!(:work) { create(:work) }
+
+    let(:email) { UserMailer.invited_to_collection_notification(otheruser.id, work.id, collection.id).deliver }
+
+    # Test the headers
+    it_behaves_like "an email with a valid sender"
+
+    it "has the correct subject line" do
+      text = "[#{ArchiveConfig.APP_SHORT_NAME}][#{collection.title}] Request to include work in a collection"
+      expect(email.subject).to eq(text)
+    end
+
+    # Test both body contents
+    it_behaves_like "a multipart email"
+    
+    it_behaves_like "a translated email"
+
+    describe "HTML version" do
+      it "has the correct content" do
+        expect(email).to have_html_part_content("would like to include your work")
+      end
+    end
+
+    describe "text version" do
+      it "has the correct content" do
+        expect(email).to have_text_part_content("would like to include your work")
+      end
+    end
+  end
+
+  describe "added to collection" do
+    let!(:gift_exchange) { create(:gift_exchange) }
+    let!(:collection) { create(:collection, challenge: gift_exchange, challenge_type: "GiftExchange") }
+    let!(:otheruser) { create(:user) }
+    let!(:work) { create(:work) }
+
+    let(:email) { UserMailer.added_to_collection_notification(otheruser.id, work.id, collection.id).deliver }
+
+    # Test the headers
+    it_behaves_like "an email with a valid sender"
+
+    it "has the correct subject line" do
+      text = "[#{ArchiveConfig.APP_SHORT_NAME}][#{collection.title}] Your work was added to a collection"
+      expect(email.subject).to eq(text)
+    end
+
+    # Test both body contents
+    it_behaves_like "a multipart email"
+
+    it_behaves_like "a translated email"
+
+    describe "HTML version" do
+      it "has text contents" do
+        expect(email).to have_html_part_content("added your work")
+      end
+      
+      it "does not have exposed HTML" do
```

Comment:
`it_behaves_like "a multipart email"` should cover checking for exposed HTML.

Reply:
Ah yes, I'd left in the other check as it spat out more useful error messages, will clean up now.